vmm/src/vm.rs

```diff
@@ -1970,16 +1970,23 @@ impl Vm {
         Some(rsdp_addr)
     }
 
-    pub fn boot(&mut self) -> Result<()> {
+    pub fn boot(&mut self, stop_vm: bool) -> Result<()> {
         info!("Booting VM");
         event!("vm", "booting");
         let current_state = self.get_state()?;
         if current_state == VmState::Paused {
             return self.resume().map_err(Error::Resume);
         }
 
-        let new_state = VmState::Running;
-        current_state.valid_transition(new_state)?;
+        let new_state = if stop_vm {
```

Comment:
Perhaps you could make the decision here rather than relying on a `stop_vm` being passed through.